fsm/README.md

```diff
@@ -0,0 +1,225 @@
+---
+layout: pattern
+title: Finite State Machine
+folder: fsm
+permalink: /patterns/state/
+categories: Behavioral
+language: en
+---
+
+## Also known as
+
+State Object
+
+## Intent
+Describe the dynamic behavior of an entity in reactive systems
+
+## Explanation
+
+Real-world example
+
+> There is a coin machine starts in a locked
+> state (Locked). When a coin is detected (Coin),
+> the machine changes to the unlocked state
+> (UnLocked) and open the turnstyle gate for
+> the person to pass. When the machine detects 
+> that a person has passed (Pass) it turns back 
+> to the locked state. If the machine is broken,
+> it is needed to be repaired.
+
+In plain words
+
+> The FSM pattern implements the state transition diagram
+> specifications such as actions, events,
+> and a state transition mechanism.
+
+*Programmatic Example**
+
+Here is the Astate(state of FSM pattern) interface and its concrete implementations.

```

Comment:
AState as a name is somewhat unclear. Translation, should it be Transition instead?